examples/elasticsearch_outlet.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import random
+import time
+from datetime import timedelta
+
+import elasticsearch
+from databay import Inlet, Link, Outlet
+from databay.inlets import RandomIntInlet
+from databay.planners import ApsPlanner
+from databay.record import Record
+
+TEXT = """
+Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+Phasellus ex erat, viverra tincidunt tempus eget, hendrerit sed ligula. 
+Quisque mollis nibh in imperdiet porttitor. Nulla bibendum lacus et est lobortis porta.
+Nulla sed ligula at odio volutpat consectetur. Sed quis augue ac magna porta imperdiet interdum eu velit. 
+Integer pretium ultrices urna, id viverra mauris ultrices ut. Etiam aliquet tellus porta nisl eleifend, non hendrerit nisl sodales. 
+Aliquam eget porttitor enim. 
+"""
+
+
+class DummyTextInlet(Inlet):
+
+    def __init__(self, text: list, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.text = text
+        self._id = 0
+
+    def pull(self, update):
+        text_selection = random.choice(self.text)
+        self._id += 1
+        time.sleep(1)
+        return {self._id: text_selection}
+
+
+class ElasticSearchIndexerOutlet(Outlet):
+    " An example outlet for indexing text documents from any `Inlet`."
+
+    def __init__(self, es_client: elasticsearch.Elasticsearch, index_name: str, overwrite_documents: bool = True):
+        super().__init__()
+        self.es_client = es_client
+        self.index_name = index_name
+
+        # if true existing documents will be overwritten
+        # otherwise we will skip indexing and log that document id exists in index.
+        self.overwrite_documents = overwrite_documents
+
+        try:
+            assert self.es_client.indices.exists(self.index_name)
```

Comment:
I know I might have used `assert` elsewhere in Databay, but I recently revised my practice on it and I think it's better to avoid it. A better choice would be an if statement and rising an explicit exception if condition is not met (ie. index doesn't exists in this case).

Either way, catching any exceptions using `except:` and raising one generic `does not exists` exception also doesn't sound like a good idea, as this may hide other exceptions raised by this third party code (eg. missing connection, bad credentials, etc.)

Reply:
Spot on have changed.